layouts/basic/modules/Vtiger/ShowAllComments.tpl

```diff
@@ -32,80 +38,48 @@
 		{/if}
 	</div>
 	{if count($HIERARCHY_LIST) != 1}
-		<div class="col-md-12 form-row commentsHeader my-3 mx-0 px-0">
-			<div class="col-md-4"></div>
-			<div class="col-md-4">
-				<div class="input-group">
-					<span class="input-group-prepend" id="commentSearchAddon">
-						<div class="input-group-text"><span class="fas fa-search"></span> </div>
-					</span>
-					<input type="text" class="form-control commentSearch" placeholder="{\App\Language::translate('LBL_COMMENTS_SEARCH','ModComments')}" aria-describedby="commentSearchAddon">
+		<div class="row">
+			<div class="col-lg-6"></div>
+			<div class="col-md-12 col-lg-6 form-row commentsHeader my-3">
+				<div class="col-6 col-lg-6 col-md-6 col-sm-6 p-0">
+					<div class="input-group-append bg-white rounded-right">
+						<input type="text" class="js-commentSearch form-control commentSearch"
+							   placeholder="{\App\Language::translate('LBL_COMMENTS_SEARCH','ModComments')}"
+							   aria-describedby="commentSearchAddon"
+							   data-js="keypress|data">
+						<button class="btn btn-outline-dark border-0 h-100 js-searchIcon searchIcon" type="button"
+								data-js="click">
+							<span class="fas fa-search fa-fw" title="{\App\Language::translate('LBL_SEARCH')}"></span>
+						</button>
+					</div>
+				</div>
+				<div class="col-5 col-lg-6 col-md-6 col-sm-6 p-0 text-right m-md-0 m-lg-0"
+					 data-toggle="buttons">
+					<div class="btn-group btn-group-toggle detailCommentsHierarchy" data-toggle="buttons">
+						<label class="btn btn-sm btn-outline-primary {if $HIERARCHY_VALUE !== 'all'}active{/if}">
+							<input class="js-detailHierarchyComments detailHierarchyComments" type="radio"
+								   name="options" id="option1"
+								   value="current" autocomplete="off"
+								   {if $HIERARCHY_VALUE !== 'all'}checked="checked"{/if}
+								   data-js="change"
+							> {\App\Language::translate('LBL_COMMENTS_0', 'ModComments')}
+						</label>
+						<label class="btn btn-sm btn-outline-primary {if $HIERARCHY_VALUE === 'all'}active{/if}">
+							<input class="js-detailHierarchyComments detailHierarchyComments" type="radio"
```

Comment:
remove detailHierarchyComments or change to BEM
**and many, many similar elements**